CMakeLists.txt

```diff
@@ -213,12 +217,16 @@ if(BML_MPI)
   add_definitions(-DDO_MPI)
 endif()
 
-include(FindCUDAToolkit)
-if(CUDAToolkit_FOUND)
-  message("Found CUDAToolkit: ${CUDAToolkit_TARGET_DIR}")
-endif()  
+if(BML_CUDA OR BML_CUSOLVER OR BML_CUSPARSE)
+  include(FindCUDAToolkit)
+  if(CUDAToolkit_FOUND)
+    message("Found CUDAToolkit: ${CUDAToolkit_TARGET_DIR}")
+  endif()
+endif()
 
-find_package(hip QUIET)
+if(BML_ROCSOLVER OR BML_ROCSPARSE)
```

Comment:
I think we can leave out this if() safely, it will just fail quietly if hip isn't found.